doc/source/examples/extended_examples/hpc/hpc_ml_ga.rst

```diff
@@ -0,0 +1,448 @@
+gradually improving the overall quality of solutions over time. Genetic algorithms are
+particularly useful for solving problems where traditional methods are impractical or
+inefficient, such as optimization, search, and machine learning tasks. They find
+applications in various fields, including engineering, economics, biology, and computer science.
+
+Problem definition
+==================
+
+This example shows how to use a generic algorithm to calculate the force
+required for a double-clamped beam to deform a specific amount in its
+center.
+
+The beam model is the same as :ref:`ref_mapdl_beam`.
+It is made of ``BEAM188`` elements that span for 2.2 meters,
+and it fully clamped at both ends.
+
+The PyMAPDL beam model is defined in the ``calculate_beam()`` function as follows:
+
+.. code-block:: python
+
+    def calculate_beam(mapdl, force):
+        # Initializing
+        mapdl.clear()
+        mapdl.prep7()
+
+        # Define an I-beam
+        mapdl.et(1, "BEAM188")
+        mapdl.keyopt(1, 4, 1)  # transverse shear stress output
+
+        # Material properties
+        mapdl.mp("EX", 1, 2e7)  # N/cm2
+        mapdl.mp("PRXY", 1, 0.27)  # Poisson's ratio
+
+        # Beam properties in centimeters
+        sec_num = 1
+        mapdl.sectype(sec_num, "BEAM", "I", "ISection", 3)
+        mapdl.secoffset("CENT")
+        mapdl.secdata(15, 15, 29, 2, 2, 1)  # dimensions are in centimeters
+
+        # Setting FEM model
+        mapdl.n(1, 0, 0, 0)
+        mapdl.n(12, 110, 0, 0)
+        mapdl.n(23, 220, 0, 0)
+        mapdl.fill(1, 12, 10)
+        mapdl.fill(12, 23, 10)
+
+        for node in mapdl.mesh.nnum[:-1]:
+            mapdl.e(node, node + 1)
+
+        # Define the boundary conditions
+        # Allow movement only in the X and Z direction
+        for const in ["UX", "UY", "ROTX", "ROTZ"]:
+            mapdl.d("all", const)
+
+        # constrain just nodes 1 and 23 in the Z direction
+        mapdl.d(1, "UZ")
+        mapdl.d(23, "UZ")
+
+        # apply a -Z force at node 12
+        mapdl.f(12, "FZ", force[0])
+
+        # run the static analysis
+        mapdl.run("/solu")
+        mapdl.antype("static")
+        mapdl.solve()
+
+        # Extracting data
+        UZ = mapdl.post_processing.nodal_displacement("Z")
+        UZ_node_12 = UZ[12]
+
+        return UZ_node_12
+
+This function returns the control parameter for the model, which is the displacement at the Z-direction on the node 12 (``UZ_node_12``).
+
+
+Setting MAPDL pool
+==================
+
+Genetic algorithms are expensive in regard to the amount of calculations needed to reach an optimal solution. As shown earlier, many simulations must be performed to select, cross over, and mutate
+all the chromosomes across all the populations.
+For this reason, to speed up the process, it is desirable to have as many MAPDL instances as possible, so
+each one can calculate one chromosome fit function.
```

Comment:
```suggestion
For this reason, to speed up the process, it is desirable to have as many MAPDL instances as possible so that each one can calculate one chromosome fit function.
```